### DISCUSSION TITLE
Dev setup

### DISCUSSION THREAD

**Discussion by lionelwust** [Instawork/hyperview#498] [Q&A] [answered]
Hello and sorry in advance for the basic question...

I am trying to run the Hyperview demo locally, like explained on the Hyperview site.

I guess I have done everything, updated constants.js, the backend server and React wrapper seem to be running, but I keep getting TypeError: Network request failed on the physical device's Expo I am running the app from.

Any obvious idea on your side ?

Thank you !!
Lionel

**Comment by adamstep** [comment]
Hi Lionel. Just to verify this step:

> Open [/demo/src/constants.js](https://github.com/Instawork/hyperview/blob/master/demo/src/constants.js) in a text editor. In ENTRY_POINT_URL, replace the host in the url (http://0.0.0.0:8085/index.xml) with the IP of your machine. This is needed in order for your physical device to be able to request the example XML files from your development machine.

I assume you found the IP. If you enter this IP/port into a web browser on your physical device, does the page load?

**Reply by lionelwust** [reply]
Hello Adam, Thank you for your answer. In a browser on my physical device, I get a timeout if I put my IP address. If I put the metro IP on port 8085, I get the XML back.

**Reply by adamstep** [reply]
I see. Whatever URL works in the browser, is the same one you should use as `ENTRY_POINT_URL` in the app. The app should render the XML at that point. Can you confirm this is what is set up in your configuration, and the physical device has the lastest code?

**Reply by lionelwust** [reply]
And of course, it works. I guess the only difference is that I tried now without the device physically connected to my computer... Before I was trying but my mobile device was usb-c linked to the computer... Anyways, thank you !

**Reply by adamstep** [reply]
Glad it works, let me know if you encounter any other issues.

**Reply by lionelwust** [reply]
Hello again,

I am back at trying Hyperview.

This time trying to setup a Django app as the backend. It is running locally on 127.0.0.1:8000, entry point is updated.

From what I see in https://hypermedia.systems/book/hyperview-a-mobile-hypermedia/, I should be getting a Parse error now since my Django app isn't returning XML yet. But I still get a TypeError: Network request failed.

Do you have any idea why ?

**Reply by adamstep** [reply]
Hi @lionelwust , when you see `Network request failed ` in the mobile app, do you see a request being made on the Django side? Generally this error means that the app did not get a response from the server. Typically this happens if something is misconfigured with regards to the entrypoint URL. Can you also confirm if you are using iOS or Android, and are you testing on a simulator/emulator or physical device?

**Reply by lionelwust** [reply]
Hello @adamstep, no I don't have any request made on Django's server. I am using a physical Android device for testing.

**Reply by adamstep** [reply, marked as answer]
Ah I see. When running on a physical device, you will need to use the IP of the computer running Django. `127.0.0.1` will not work across devices since it's a special localhost address. The instructions for how to find our your computer's IP depend on the Platform (Windows, Mac, Linux). Once you find it, if you use it in `ENTRY_POINT_URL`, then the request should work.

**Reply by lionelwust** [reply]
Yes, now I see a ParseError. Good, thank you very much for your help.

**Reply by adamstep** [reply]
Glad that worked. Let me know if you have any other questions, we use Django with Hyperview so I'm sure I've encountered every potential gotcha :)

**Reply by lionelwust** [reply]
I trust you on that 😄 One question that I would have right away is: do you sometimes use the same views for web and mobile. If yes, how to you differentiate in order to return html or hxml ?

I was wondering about having a specific set of urls for mobile, but potentially calling the same views. Or setting a header from the React native app (i don't know how to do it yet). Or maybe you have a better way
